### PR TITLE
Added a clear message about failed CORS validation requests (close #2482)

### LINKED ISSUE #2482: [RequestMock] Add a warning with a clear message when CORS validation is failed
I'm testing to mock a remote API endpoint (http://remote-api.com/api/auth?user_type=customer) that is being called with "isomorphic-fetch" when filling a form and clicking in a simple submit button. It seems that the mock is returning HTTP status 222. Does this means it can be a CORS issue? Or a bug of the RequestMock?

Tested page URL: http://localhost:3000/

Test code

```js
const mock = RequestMock()
  .onRequestTo('http://remote-api.com/api/auth?user_type=customer')
  .respond({
    result: null,
    errors: [
      {
        message: 'Incorrect username or password.',
      },
    ],
  });

fixture('As a customer I want to be able to sign in')
  .page('http://localhost:3000/login')
  .requestHooks(mock);

test('Login with incorrect credentials', async (t) => {
  await t
    .typeText(Selector("input[name='email']"), 'joe@example.com')
    .typeText(Selector("input[name='password']"), 'password')
    .click(Selector("input[type='submit']"));
  const location = await t.eval(() => window.location);
  await t.expect(location.pathname).eql('/login');
});

```

* operating system: Mac OS X 10.13.4
* testcafe version: 0.20.0
* node.js version: 8.11.1

### PR DESCRIPTION
**Pull request by miherlosev**
@AndreyBelym 

The text of the warning was checked by technical writers.

### CODE REVIEW

File: test/functional/fixtures/api/es-next/request-hooks/test.js [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+    describe('RequestMock', () => {
+        it('Basic', () => {
+            return runTests('./testcafe-fixtures/request-mock/basic.js', 'Basic', { only: 'chrome' });
+        });
+
+        it('Request failed the CORS validation', () => {
+            return runTests('./testcafe-fixtures/request-mock/failed-cors-validation.js', 'Failed CORS validation', { only: 'chrome' })
+                .then(() => {
+                    expect(testReport.warnings).eql([
+                        'RequestMock: CORS validation failed for a request specified as { url: "http://dummy-url.com/get" }'
```
Comment (AndreyBelym) [inline test/functional/fixtures/api/es-next/request-hooks/test.js]: Can we report the fixture name and the test name where the warning emitted?

**Comment by AndreyBelym** [comment]
Could you please explain why you decided to not throw an error and emit a warning instead?

File: test/functional/fixtures/api/es-next/request-hooks/test.js (hunk 1 above)
Comment (miherlosev) [reply test/functional/fixtures/api/es-next/request-hooks/test.js]: For all [warning messages](https://github.com/DevExpress/testcafe/blob/master/src/notifications/warning-message.js) we don't report fixture and test name where the warning was emitted.

We can plan this as an enhancement.
Also the warning message contains stringified request filter rule that cause the warning.

**Comment by miherlosev** [comment]
This case should not stop a test execution. So, it is not an error.

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
FPR

**Pull request merged**